### PR TITLE
java: fix memory leak in tests

### PR DESCRIPTION
**Pull request by matklad**
NativeClient holds onto native memory, it should be deleted explicitly!

### CODE REVIEW

File: src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java [hunk 1]
```diff
@@ -74,473 +73,479 @@ public void testConstructorWithClientNull() {
...
     @Test
     public void testEndRequestWithNullBuffer() throws Throwable {
+        try (var client = getDummyClient()) {
+            var batch = new TransferBatch(1);
+            batch.add();
 
-        var client = getDummyClient();
-        var batch = new TransferBatch(1);
-        batch.add();
+            var callback = new CallbackSimulator<CreateTransferResultBatch>(
```
Comment (matklad) [inline src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java:164]: Syntax for C#'s `using` is absolutely genius!

File: src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java [hunk 2]
```diff
@@ -20,51 +20,50 @@ public class AsyncRequestTest {
 
     @Test
     public void testCreateAccountsRequestConstructor() {
+        try (var client = getDummyClient()) {
```
Comment (batiati) [inline src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java:23]: Why don't we use a single static client here, like we did for the Integration tests?

**Review by batiati — COMMENTED** [review]
🙌

File: src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java (hunk 2 above)
Comment (matklad) [reply src/clients/java/src/test/java/com/tigerbeetle/AsyncRequestTest.java:23]: We _don't_ need single client here, as echoClient is cheap and we don't need to amortise. And with many clients, we exercise open/close code path more, which seems like a good thing. With a single client we probably wouldn't have noticed the leak here at all.

**Review by batiati — APPROVED** [review]

**Pull request merged**